wsaa.py

```diff
@@ -188,32 +219,46 @@ def sign_tra(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
         else:
             raise RuntimeError("Part not found")
 
-    else:
-        # Firmar el texto (tra) usando OPENSSL directamente
-        try:
-            out = Popen(
-                [
-                    openssl_exe(),
-                    "smime",
-                    "-sign",
-                    "-signer",
-                    cert,
-                    "-inkey",
-                    privatekey,
-                    "-outform",
-                    "DER",
-                    "-nodetach",
-                ],
-                stdin=PIPE,
-                stdout=PIPE,
-                stderr=PIPE,
-            ).communicate(tra)[0]
-            return b64encode(out)
-        except OSError as e:
-            if e.errno == 2:
-                warnings.warn("El ejecutable de OpenSSL no esta disponible en el PATH")
-            raise
 
+def sign_tra_openssl(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
+    "Workaround using openssl binary directly via command-line interface to sign transactions"
+    try:
+        out = Popen(
+            [
+                openssl_exe(),
+                "smime",
+                "-sign",
+                "-signer",
+                cert,
+                "-inkey",
+                privatekey,
+                "-outform",
+                "DER",
+                "-nodetach",
+            ],
+            stdin=PIPE,
+            stdout=PIPE,
+            stderr=PIPE,
+        ).communicate(tra)[0]
+        return b64encode(out)
+    except OSError as e:
+        if e.errno == 2:
+            warnings.warn("El ejecutable de OpenSSL no esta disponible en el PATH")
+        raise
+
+
+def sign_tra(tra, cert=CERT, privatekey=PRIVATEKEY, passphrase=""):
+    "Firmar PKCS#7 el TRA y devolver CMS (recortando los headers SMIME)"
+
+    if isinstance(tra, str):
+        tra = tra.encode("utf8")
+
+    if Binding:
+        if sys.version_info.major == 2:
```

Comment:
This also could be needed in Python 3 if using outdated libraries.
The condition should be rewritten to check cryptography version < 39